src/commondao/common.dao.ts

```diff
@@ -600,19 +613,34 @@ export class CommonDao<
     return bm as any
   }
 
-  /**
-   * Mutates id if needed
-   */
+  private async ensureImmutableDoesntExist(table: string, dbm: DBM): Promise<void> {
+    await this.throwIfObjectExists(
+      table,
+      dbm,
+      new AppError(DBLibError.OBJECT_IS_IMMUTABLE, {
```

Comment:
I see that you made it DRY. This looks good.

But I have one concern about performance of `new Error`. I recently benchmarked code that does `new Error` vs code without it, and the perf difference was ~1000 times (because new Error traverses many files, reads source code to build stack trace, etc). So, I'd prefer to not do `new Error` for "happy cases" where we might NOT throw it. It'll be less DRY (just the error creation part), but more performant.

Reply:
Just repeated my benchmark to refresh my memory. Ok, it's not 1000 times but 10 times instead. But still notable. Benchmark lives in `js-lib`.

<img width="554" alt="image" src="https://user-images.githubusercontent.com/4757745/153420734-f6b96690-5166-466e-8534-9c6544358a76.png">